pyfstat/make_sfts.py

```diff
@@ -334,8 +340,14 @@ def _get_setup_from_timestamps(self):
         self.sftfilenames = []  # This refers to the MFD output!
         for X, IFO in enumerate(IFOs):
             tsX = np.genfromtxt(tsfiles[X], comments="%")
-            this_start_time = int(tsX[0, 0])
-            this_end_time = int(tsX[-1, 0]) + self.Tsft
+            if len(np.shape(tsX)) > 1:
+                logger.warning(
+                    f"Timestamps file {tsfiles[X]} has more than 1 column,"
+                    " we will ignore the rest."
```

Comment:
Opening discussion:

Most LALSuite timestamp files have two columns ((please correct me if this is just my case). Should this really be a bare warning?

As per what we discussed:
1. Two column files with wanoseconds, ignoring nanoseconds --> Not really a surprise, since CW software does that anyways.
2. Two column files with second column being zeros --> Not really a surprise either.
3. Two column files which are actually *segment* files --> Should crash, that's not how it works.

Q: Is there any simple way of distinguishing 1. and 3.?

Reply:
> Most LALSuite timestamp files have two columns ((please correct me if this is just my case)

https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___s_f_tfile_i_o__h.html#gaaf70ab5d23c42df41f76e2702cfe1098

-> the seconds,nanoseconds format is actually considered **obsolete** as far as lalpulsar is concerned

That doesn't technically contradict your point though - in more than just this one case, _most_ of what people do is legacy nowadays... :wink:

Regarding the segments format, the only way to identify (most of) them would be by saying that (recent) GPS seconds are one digit longer than what nanosecond remnants should be. Anyway, there are common files of this type with one SFT per line/segment (`ts[n,1]-ts[n,0]=Tsft`), so I wouldn't hard-fail on those. As long as the docs are clear what we do with the files, users who give us weird file formats will have the responsibility to check if they got back the right results.